mbedtls/src/ssl/context.rs

```diff
@@ -70,13 +71,7 @@ impl<IO: Read + Write + 'static> IoCallback for IO {
 define!(
     #[c_ty(ssl_context)]
     #[repr(C)]
-    struct Context {
-        // config is used read-only for mutliple contexts and is immutable once configured.
-        config: Arc<Config>, 
-
-        // Must be held in heap and pointer to it as pointer is sent to MbedSSL and can't be re-allocated.
-        io: Option<Box<dyn Any>>,
-        
+    struct HandshakeContext {
```

Comment:
Please add a test that this type is !Sync. I think we don't expose owned values, so checking !Send isn't necessary.